tools/common-code/cmonitor_launcher.py

```diff
@@ -0,0 +1,195 @@
+# =======================================================================================================
+# cmonitor_launcher.py
+#
+# Author: Satyabrata Bharati
+# Created: April 2022
+#
+# =======================================================================================================
+
+import inotify.adapters
+import concurrent.futures
+from concurrent.futures import ProcessPoolExecutor
+from subprocess import Popen
+import queue
+import os
+import time
+
+queue = queue.Queue()
+# =======================================================================================================
+# CmonitorLauncher
+#
+#  - Watch all files below a directory and notify an event for changes.
+#  - Retrieves all the process and extract the process name "/proc/<pid>/stat.
+#  - check the process name against the white-list given in the filter list.
+#  - Execute command to launch CMonitor if the process name matches with the filter.
+#
+# =======================================================================================================
+class CmonitorLauncher:
```

Comment:
this class contains both the logic to watch for inotify events and to process them. We must split these 2 parts into 2 different classes:
1) CgroupWatcher : this class will contain the inotify_events() method but it should also provide more "value" by containing the white filter logic on the process name; in particular it should insert in the "queue" only when an event is matching all filtering criterias (to avoid unblocking "by mistake" the worker thread)

2) CmonitorLauncher : this class will contain the launch_cmonitor() method. I think using os.system() may raise subtle issues: for example if cmonitor_launcher.py is killed, there is no way for this script to terminate all the children cmonitor processes as well. By using instead subprocess.popen() or maybe even with subprocess.run() it should be instead possible to propagate the SIGTERM to all children cmonitor instances.
Anyway this is something that needs to be tested properly (maybe SIGTERM works also with os.system() not sure!!!)